### PR TITLE
fix(test): Handle CPU feature diff between host and guest correctly

### PR DESCRIPTION
**Pull request by zulinx86**
## Reason

PR #4884 introduced a test that ensures CPU feature diff between host and guest hasn't been changed.

However, the expected diff can vary depending on a combination of host kernel version and guest kernel version.

The current our setting is as follows:
- Host
  - Amazon Linux 2 (v5.10)
  - Amazon Linux 2023 (v6.1)
  - Ubuntu 24.04 (v6.8)
- Guest
  - Amazon Linux microvm kernel (v5.10)
  - Amazon Linux microvm kernel (v6.1)

## Changes

- aarch64
  - Add "ssbs" to the expected diff if needed.
- x86_64
  - Remove "flush_l1d" from the expected diff if the host passes through it.
  - Add "invpcd_single" to the expected diff if needed.

**Note that I implemented it in a general way as possible because we might want to support more host and guest kernels in the future.**

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this
  PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- ~~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/functional/test_cpu_features_aarch64.py [hunk 1]
```diff
@@ -82,11 +83,16 @@ def test_host_vs_guest_cpu_features_aarch64(uvm_nano):
...
         case CpuModel.ARM_NEOVERSE_N1:
             assert host_feats - guest_feats == set()
-            # Kernel should hide this feature, but our guest kernel
-            # currently lacks the commit with this change.
-            # The commit that introduces the change:
-            # https://github.com/torvalds/linux/commit/7187bb7d0b5c7dfa18ca82e9e5c75e13861b1d88
-            assert guest_feats - host_feats == {"ssbs"}
+            # TODO: Remove "ssbs" once our guest kernel picks the following commit.
+            # https://github.com/torvalds/linux/commit/adeec61a4723fd3e39da68db4cc4d924e6d7f641
+            # See also https://github.com/firecracker-microvm/firecracker/issues/4898
```
Comment (pb8o) [inline tests/integration_tests/functional/test_cpu_features_aarch64.py:88]: So for the feature to disappear, both host and guest kernel need the commit?

File: tests/integration_tests/functional/test_cpu_features_x86_64.py [hunk 2]
```diff
@@ -392,6 +392,15 @@ def test_host_vs_guest_cpu_features_x86_64(uvm_nano):
                 "vpid",
                 "xtpr",
             }
+            # TODO: Remove the below ubuntu exception and "flush_l1d" from the above expected list
+            # once Amazon Linux 2 and Amazon Linux 2023 picks the following kernel commit and passes
+            # through the "flush_l1d" CPUID bit to guests.
+            # https://github.com/torvalds/linux/commit/45cf86f26148e549c5ba4a8ab32a390e4bde216e
+            # See also https://github.com/firecracker-microvm/firecracker/issues/4899
+            if "Ubuntu" in global_props.os:
```
[the remark this answers is not in the document]
Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_features_x86_64.py:400]: See https://github.com/firecracker-microvm/firecracker/issues/4899.

File: tests/integration_tests/functional/test_cpu_features_aarch64.py (hunk 1 above)
Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_features_aarch64.py:88]: yes. Currently only the host has the change that removes ssbs from the feature flags.

File: tests/integration_tests/functional/test_cpu_features_x86_64.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pb8o) [reply tests/integration_tests/functional/test_cpu_features_x86_64.py:400]: Could we then make the check:

```
if global_props.linux_version_tpl >= (6, 8)
```

**Review by ShadowCurse — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**